azure-keyvault/azure/keyvault/keys/_client.py

```diff
@@ -58,236 +44,552 @@ def create_config(**kwargs):
+        request = HttpRequest("POST", url, headers, data=request_body)
+        request.format_parameters(query_parameters)
 
         response = self._pipeline.run(request, **kwargs).http_response
         if response.status_code != 200:
-            raise ClientRequestError(
-                "Request failed with code {}: '{}'".format(
-                    response.status_code, response.text()
-                )
-            )
+            raise ClientRequestError("Request failed status code {}.  {}".format(response.status_code, response.text()))
 
-        key = self._deserialize("Key", response)
+        bundle = DESERIALIZE("KeyBundle", response)
 
-        return key
+        return Key._from_key_bundle(bundle)
 
     def delete_key(self, name, **kwargs):
-        # type: (str, Any) -> DeletedKey
+        # type: (str, Mapping[str, Any]) -> DeletedKey
+        """Deletes a key of any type from storage in Azure Key Vault.
+
+        The delete key operation cannot be used to remove individual versions
+        of a key. This operation removes the cryptographic material associated
+        with the key, which means the key is not usable for Sign/Verify,
+        Wrap/Unwrap or Encrypt/Decrypt operations. This operation requires the
+        keys/delete permission.
+
+        :param name: The name of the key to delete.
+        :type name
+        :returns: The deleted key
+        :rtype: ~azure.keyvault.keys._models.DeletedKey
+        :raises: ~azure.core.exceptions.ClientRequestError if the client failed to delete the key
+
+        Example:
+            .. literalinclude:: ../tests/test_examples_keys.py
+                :start-after: [START delete_key]
+                :end-before: [END delete_key]
+                :language: python
+                :dedent: 4
+                :caption: Deletes a key in the key vault
+        """
         url = "/".join([self.vault_url, "keys", name])
 
         request = HttpRequest("DELETE", url)
-        request.format_parameters({"api-version": self.API_VERSION})
+        query_parameters = {"api-version": self._api_version}
+        request.format_parameters(query_parameters)
+
         response = self._pipeline.run(request, **kwargs).http_response
         if response.status_code != 200:
-            raise ClientRequestError(
-                "Request failed with code {}: '{}'".format(
-                    response.status_code, response.text()
-                )
-            )
-        deleted_key = self._deserialize("DeletedKey", response)
+            raise ClientRequestError("Request failed status code {}.  {}".format(response.status_code, response.text()))
+
+        bundle = DESERIALIZE("DeletedKeyBundle", response)
 
-        return deleted_key
+        return DeletedKey._from_deleted_key_bundle(bundle)
 
-    def get_key(self, name, version="", **kwargs):
-        # type: (str, str, Any) -> Key
+    def get_key(self, name, version, **kwargs):
```

Comment:
>version, [](start = 28, length = 8)

version needs to be an optional parameter.

Reply:
For secrets as well?